sidecar/internal/util/util.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2022 The Kubernetes-CSI-Addons Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+)
+
+// ValidateControllerEndpoint validate endpoint to make sure it contains
+// valid ip address and port number.
+func ValidateControllerEndpoint(endpoint string) error {
+	separator := ":"
+	splits := strings.Split(endpoint, separator)
+	if len(splits) != 2 {
+		return fmt.Errorf("invalid endpoint %q, expected format '<ip>:<port>'", endpoint)
+	}
+
+	rawIP, rawPort := splits[0], splits[1]
+
+	ip := net.ParseIP(rawIP)
+	if ip == nil {
+		return fmt.Errorf("invalid ip address %q in endpoint %q", rawIP, endpoint)
+	}
+
+	_, err := strconv.ParseUint(splits[1], 10, 16)
```

Comment:
max port number validation is required?

Reply:
> max port number validation is required?

I was not sure about that, but while creating pod and exposing a port, the port is validated, so I guess we don't need to have that check here.